app/models/user.rb

```diff
@@ -0,0 +1,29 @@
+require 'bcrypt'
+
+class User
+  include DataMapper::Resource
+  include BCrypt
+  attr_reader :password
+
+  property :id, Serial
+  property :username, String
+  property :email, String
+  property :name, String
+  property :password_hash, Text
+
+  def password=(password)#overrides the <property name>= method but only for properties named 'password.' So rather than saving it under a password column in the db, the pw is hashed and transferred to a password_hash column
```

Comment:
Metrics/LineLength: Line is too long. [224/90] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)